src/bitdrift_public/fbs/issue-reporting/v1/report.fbs

```diff
@@ -0,0 +1,300 @@
+    // A unique identifier for the app, such as BuildConfig.APPLICATION_ID on
+    // Android or CFBundleIdentifier from the Info.plist on Apple platforms.
+    app_id:string;
+
+    // Memory state at the moment the event occurred
+    memory:Memory;
+
+    // The installed app version, such as BuildConfig.VERSION_NAME on Android
+    // or CFBundleShortVersionString from the Info.plist on Apple platforms.
+    version:string;
+
+    // Build number (platform-specific names and types)
+    build_number:AppBuildNumber;
+}
+
+table OSBuild {
+    // Current version of the device's operating system
+    version:string;
+
+    // (Android-specific) Consumer-visible brand
+    // See: https://developer.android.com/reference/android/os/Build#BRAND
+    brand:string;
+
+    // (Android-specific) Unique identifier for the build itself
+    // See: https://developer.android.com/reference/android/os/Build#FINGERPRINT
+    fingerprint:string;
+
+    // (Apple platforms-specific) Identifier differentiating different builds
+    // within a particular OS version. Most helpful during Beta Release Season.
+    kern_osversion:string;
+}
+
+enum Rotation : int8 {
+    Unknown = 0,
+
+    // Natural orientation. Rotated 0º
+    Portrait,
+
+    // "Bottom" of the device pointed to the right. Rotated 90º
+    LandscapeRight,
+
+    // "Bottom" of the device pointed to the left. Rotated 270º
+    LandscapeLeft,
+
+    // Rotated 180º
+    PortraitUpsideDown,
+}
+
+table PowerMetrics {
+    power_state:PowerState;
+
+    // Amount charged, number between 0 and 100
+    charge_percent:uint8;
+}
+
+table DeviceMetrics {
+    // The moment the event occurred
+    time: Timestamp;
+
+    timezone:string;
+
+    power_metrics:PowerMetrics;
+
+    network_state:NetworkState;
+
+    rotation:Rotation;
+
+    arch:Architecture;
+
+    manufacturer:string;
+    model:string;
+
+    os_build:OSBuild;
+
+    // Development platform or operating system manufacturer
+    platform:Platform;
+
+    // Android-specific, ABIs supported by the device, in order of preference
+    // See: https://developer.android.com/reference/android/os/Build#SUPPORTED_ABIS
+    cpu_abis:[string];
+}
+
+table SourceFile {
+    path:string;
+    line:int64;
+    column:int64;
+}
+
+table Frame {
+    type:FrameType;
+
+    // the fully-qualified class name, if any
+    class_name:string;
+
+    // the method or function name
+    symbol_name:string;
+
+    // source file path
+    source_file:SourceFile;
+
+    // identifier for the compiled binary originating this line of code
+    image_id:string;
+
+    // address of the current execution frame
+    frame_address:uint64;
+
+    // address of the named symbol nearest above the current execution frame
+    symbol_address:uint64;
+
+    // platform-specific context explaining the thread state, if any. Examples:
+    // - locked resource causing thread blocked state
+    // - awaited thread ID
+    state:[string];
+}
+
+table Thread {
```

Comment:
would we ever want to expose register values per thread? this is something the tombstone exposes